control-plane/plugin/src/resources/pool.rs

```diff
@@ -60,18 +61,77 @@ impl GetHeaderRow for openapi::models::Pool {
     }
 }
 
+#[derive(Debug, Clone, clap::Args)]
+/// Arguments used when getting nodes.
+pub struct GetPoolsArgs {
+    /// Id of the node
+    #[clap(long)]
+    node_id: Option<NodeId>,
+
+    /// Filter pools on labels
+    #[clap(long)]
+    labels: Option<String>,
+}
+
+impl GetPoolsArgs {
+    /// Return the node ID.
+    pub fn node_id(&self) -> Option<NodeId> {
+        self.node_id.clone()
+    }
+    /// Filter on the labels provided
+    pub fn filter_labels(&self) -> Option<String> {
+        self.labels.clone()
+    }
+}
+
 #[async_trait(?Send)]
-impl List for Pools {
-    async fn list(output: &utils::OutputFormat) -> PluginResult {
-        match RestClient::client().pools_api().get_pools().await {
-            Ok(pools) => {
-                // Print table, json or yaml based on output format.
-                utils::print_table(output, pools.into_body());
+impl ListWithArgs for Pools {
+    type Args = GetPoolsArgs;
+    async fn list(args: &Self::Args, output: &utils::OutputFormat) -> PluginResult {
+        let pools = match args.node_id() {
+            Some(node_id) => {
+                match RestClient::client()
+                    .pools_api()
+                    .get_node_pools(&node_id)
+                    .await
+                {
+                    Ok(pools) => pools.into_body(),
+                    Err(e) => {
+                        return Err(Error::ListPoolsError { source: e });
```

Comment:
Add Ok above and use ? on 110.
```suggestion
                        Err(Error::ListPoolsError { source: e });
```